gensim/models/translation_matrix.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env python
+# encoding: utf-8
+import warnings
+import numpy as np
+
+from collections import OrderedDict
+from gensim import utils
+from six import string_types
+
+
+"""
+Produce translation matrix to translate the word from one language to another language, using either
+standard nearest neighbour method or globally corrected neighbour retrieval method [1].
+
+This method can be used to augment the existing phrase tables with more candidate translations, or
+filter out errors from the translation tables and known dictionaries [2].
+
+Initialize a model with e.g.::
+
+    >>> transmat = TranslationMatrix(word_pair, source_word_vec, target_word_vec)
+
+Train a model with e.g.::
+
+    >>> transmat.train(word_pair)
+
+Persist a model to disk with::
+
+    >>> transmat.save(fname)
+    >>> transmat  = TranslationMatrix.load(fname)
+
+Translate the source words to target words, for example
+
+    >>> transmat.translate(["one", "two", "three"], topn=3)
+
+.. [1] Dinu, Georgiana, Angeliki Lazaridou, and Marco Baroni. "Improving zero-shot learning by mitigating the
+        hubness problem." arXiv preprint arXiv:1412.6568 (2014).
+.. [2] Tomas Mikolov, Ilya Sutskever, Kai Chen, Greg Corrado, and Jeffrey Dean. Distributed Representations of Words and Phrases and their Compositionality.
+       In Proceedings of NIPS, 2013.
+"""
+
+
+class Space(object):
+    """
+    An auxiliary class for storing the the words space
+
+    Attributes:
+        `mat` (ndarray): each row is the word vector of the lexicon
+        `index2word` (list): a list of lexicon
```

Comment:
Terms like 'lexicon' unclear: is `index2word` a list of strings?

Reply:
Yes, index2word is a list of words(strings).